src/core/federation.ts

```diff
@@ -835,7 +843,16 @@ export class FederatedApp extends App {
             let greatestTAG = this._getGreatestTimeAdvanceGrant();
             let nextTag = event.tag;
             if (greatestTAG === null || greatestTAG.isSmallerThan(nextTag)) {
-                this.sendRTINextEventTime(nextTag);
+                if (this.minDelayFromPhysicalActionToFederateOutput !== null &&
+                    this.downstreamFedIDs.length > 0) {
+                        let physicalTime = getCurrentPhysicalTime()
+                        if (physicalTime.add(this.minDelayFromPhysicalActionToFederateOutput).isEarlierThan(event.tag.time)) {
+                            Log.debug(this, () => "Adding dummy event for tag: " + event.tag);
+                            this._addDummyEvent(event.tag);
```

Comment:
One question: shouldn't this be `this._addDummyEvent(physicalTime.add(this.minDelayFromPhysicalActionToFederateOutput))`?

Reply:
Thank you for catching this! This PR https://github.com/lf-lang/reactor-ts/pull/109 should address this issue.